packages/front-end/components/Segments/SegmentForm.tsx

```diff
@@ -85,30 +85,37 @@ const SegmentForm: FC<{
           })}
         />
       )}
-      <Field
-        label={sql ? "SQL" : "Event Condition"}
-        required
-        textarea
-        {...form.register("sql")}
-        placeholder={
-          sql
-            ? `SELECT ${userIdType}, date FROM mytable`
-            : "event.properties.$browser === 'Chrome'"
-        }
-        helpText={
-          sql ? (
+      {sql ? (
```

Comment:
Originally, we had a single `<Field />` component, and we had a bunch of ternaries inside to determine things like `placeholder`, `label`, etc. I initially went down that path, but it got suuuper complicated/hard to read, so I broke it out like this. It's a bit longer and means there is some repeating, but I feel it's 100X easier to read. Open to other ideas and suggestions, though.